### PR TITLE
Shiny module

### PR DESCRIPTION
**Pull request by jcheng5**
Adds a profvis Shiny module, for starting/stopping the profiler during the execution of a Shiny application. This can be helpful if you don't want to profile the entire execution of an app, only a certain operation. To install the profvis module into your app, add `profvis_ui("profvis")` to your UI, and `callModule(profvis_server, "profvis")` to your server function.

Credits: Implemented by @tmastny, possibly @wch helped too.

The profvis module is the little widget at the top/center in the following screenshots. (The colors may look slightly ugly, that's just the Bootstrap 3 theme in force for this app.)

image
image
image

## Testing notes

To run a (very) simple example:

```r
library(profvis)
example(profvis_ui, ask=FALSE)
```

### CODE REVIEW

**Comment by jcheng5** [comment]
This would be even better if hitting "Stop Recording" immediately launched a new tab with the profvis visualization for the newly saved recording.

**Comment by jcheng5** [comment]
"Stop Recording" now launches a new tab.

File: R/shiny_module.R [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+#' Rprof output. Used with \code{\link{profvis_ui}}. This function requires the
+#' \code{shiny} package.
+#'
+#' @param input,output,session Arguments used by
+#'   \code{\link[shiny]{callModule}}.
+#' @param dir Output directory to save Rprof files.
+#'
+#' @examples
+#' if(interactive()) {
+#'   shinyApp(
```
Comment (schloerke) [inline R/shiny_module.R]: Add a `library(shiny)`

File: R/shiny_module.R [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+#'       plotOutput("plot"),
+#'       actionButton("new", "New plot"),
+#'       profvis_ui("profiler")
+#'     ),
+#'     function(input, output, session) {
+#'       callModule(profvis_server, "profiler")
+#'
+#'       output$plot <- renderPlot({
+#'         input$new
+#'         ggplot(diamonds, aes(carat, price)) + geom_point()
```
Comment (schloerke) [inline R/shiny_module.R]: Need a `library(ggplot2)` at the top

File: R/shiny_module.R [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
```
Comment (schloerke) [inline R/shiny_module.R]: Need a `library(ggplot2)` at the top

File: R/shiny_module.R [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+#' a profvis interface to profile parts of a shiny app and save the profvis or
+#' Rprof output. Used with \code{\link{profvis_ui}}. This function requires the
+#' \code{shiny} package.
+#'
+#' @param input,output,session Arguments used by
+#'   \code{\link[shiny]{callModule}}.
+#' @param dir Output directory to save Rprof files.
+#'
+#' @examples
+#' if(interactive()) {
```
Comment (schloerke) [inline R/shiny_module.R]: If this example is the exact same as `profvis_ui`, could call `example("profvis_ui", "profvis")`

Comment (schloerke) [reply R/shiny_module.R]: (^^ Style choice)

**Comment by jcheng5** [comment]
@schloerke This should be ready for you

**Comment by schloerke** [comment]
@jcheng5 approved, knowing that when you call `examples(profvis_ui)` it requires interaction to produce the plots.

```
> example(profvis_ui)

prfvs_> if(interactive()) {
prfvs_+   library(shiny)
prfvs_+   library(ggplot2)
prfvs_+   shinyApp(
prfvs_+     fluidPage(
prfvs_+       plotOutput("plot"),
prfvs_+       actionButton("new", "New plot"),
prfvs_+       profvis_ui("profiler")
prfvs_+     ),
prfvs_+     function(input, output, session) {
prfvs_+       callModule(profvis_server, "profiler")
prfvs_+ 
prfvs_+       output$plot <- renderPlot({
prfvs_+         input$new
prfvs_+         ggplot(diamonds, aes(carat, price)) + geom_point()
prfvs_+       })
prfvs_+     }
prfvs_+   )
prfvs_+ }

Listening on http://127.0.0.1:5527
Hit <Return> to see next plot: 
Hit <Return> to see next plot: 
```

If you're ok with user being required to hit enter, then it's good.

File: R/shiny_module.R [hunk 5]
```diff
@@ -0,0 +1,210 @@
+#' Profvis UI for Shiny Apps
+#'
+#' Use this Shiny module to inject Profvis controls into your Shiny app. The
```
Comment (hadley) [inline R/shiny_module.R]: This is a bit late but the name of the package is profvis, not Profvis

**Pull request merged**